### PR TITLE
[OPENJPA-2744] commons-pool is updated to the most recent version

### PR DESCRIPTION
**Pull request by solomax**

### CODE REVIEW

File: openjpa-all/pom.xml [hunk 1]
```diff
@@ -31,7 +31,6 @@
         <version>3.0.1-SNAPSHOT</version>
     </parent>
 
-    <groupId>org.apache.openjpa</groupId>
```
Comment (ilgrosso) [inline openjpa-all/pom.xml]: This change is not related to the topic, I don't see the reason to include it here, please revert.

File: openjpa-kernel/src/main/java/org/apache/openjpa/event/TCPRemoteCommitProvider.java [hunk 2]
```diff
@@ -782,15 +816,17 @@ private HostAddress(String host)
...
-            _socketPool = new GenericObjectPool
-                (new SocketPoolableObjectFactory(), _maxActive,
-                    GenericObjectPool.WHEN_EXHAUSTED_BLOCK, -1);
+            _socketPool = new GenericObjectPool<>(new SocketPoolableObjectFactory(), cfg);
             _isAvailable = true;
         }
 
         private void setMaxActive(int maxActive) {
-            _socketPool.setMaxActive(maxActive);
+            _socketPool.setMaxTotal(maxActive);
```
Comment (ilgrosso) [inline openjpa-kernel/src/main/java/org/apache/openjpa/event/TCPRemoteCommitProvider.java]: Why this change? Is `setMaxActive()`not available anymore?

File: openjpa-all/pom.xml (hunk 1 above)
Comment (solomax) [reply openjpa-all/pom.xml]: Maven raises the warning: `groupId` is duplicated
I'll move all these changes to separate PR

File: openjpa-kernel/src/main/java/org/apache/openjpa/event/TCPRemoteCommitProvider.java (hunk 2 above)
Comment (solomax) [reply openjpa-kernel/src/main/java/org/apache/openjpa/event/TCPRemoteCommitProvider.java]: Yep, it is replaced with `setMaxTotal`

Comment (ilgrosso) [reply openjpa-kernel/src/main/java/org/apache/openjpa/event/TCPRemoteCommitProvider.java]: I see: would then make sense to rename the enclosing method accordingly?

Comment (solomax) [reply openjpa-kernel/src/main/java/org/apache/openjpa/event/TCPRemoteCommitProvider.java]: I believe this is the decision OpenJPA dev's need to make ...
I tried to make changes minimal.

Comment (ilgrosso) [reply openjpa-kernel/src/main/java/org/apache/openjpa/event/TCPRemoteCommitProvider.java]: Ok, I was comparing [1.6](https://commons.apache.org/proper/commons-pool/api-1.6/org/apache/commons/pool/impl/GenericObjectPool.html#setMaxActive(int))'s with [2.6.0](https://commons.apache.org/proper/commons-pool/api-2.6.0/index.html)'s javadocs and the description is effectively the same.

**Comment by ilgrosso** [comment]
Thanks again @solomax !

Next targets could be:
* commons-collections -> commons-collections4
* commons-dbcp -> commons-dbcp2

WDYT? ;-)

**Comment by solomax** [comment]
Already working on `commons-collections -> commons-collections4`
Found one serious issue so far, please check https://markmail.org/message/l72ev2vezesgbfj2
Maybe you can suggest something ...
